server/plugin/api.go

```diff
@@ -984,10 +987,29 @@ func (p *Plugin) getYourAssignments(c *UserContext, w http.ResponseWriter, r *ht
 	result, _, err := githubClient.Search.Issues(c.Ctx, query, &github.SearchOptions{})
 	if err != nil {
 		c.Log.WithError(err).With(logger.LogContext{"query": query}).Warnf("Failed to search for assignments")
-		return
+		return nil
 	}
 
-	p.writeJSON(w, result.Issues)
+	return result.Issues
+}
+
+func (p *Plugin) getSidebarData(c *UserContext) *SidebarContent {
+	assignments := p.getYourAssignmentsData(c)
+	prs := p.getYourPrsData(c)
+	reviews := p.getReviewsData(c)
+	unreads := p.getUnreadsData(c)
+
+	return &SidebarContent{
+		Assignments: assignments,
+		PRS:         prs,
+		Reviews:     reviews,
+		Unreads:     unreads,
+	}
+}
```

Comment:
```suggestion
func (p *Plugin) getSidebarData(c *UserContext) *SidebarContent {
	return &SidebarContent{
		Assignments: p.getYourAssignmentsData(c),
		PRS:         p.getYourPrsData(c),
		Reviews:     p.getReviewsData(c),
		Unreads:     p.getUnreadsData(c),
	}
}
```
see if we can we do this